addons/website_event/static/src/interactions/website_event_slot_details.js

```diff
@@ -0,0 +1,141 @@
+import { Interaction } from "@web/public/interaction";
+import { registry } from "@web/core/registry";
+import { rpc } from "@web/core/network/rpc";
+import { _t } from "@web/core/l10n/translation";
+
+const { DateTime } = luxon;
+
+export class SlotDetails extends Interaction {
+    static selector = ".o_wevent_js_slot_details";
+    dynamicSelectors = {
+        ...this.dynamicSelectors,
+        _envBus: () => this.env.bus,
+    };
+    dynamicContent = {
+        _envBus: {
+            "t-on-websiteEvent.enableSubmit": () => this.buttonDisabled = false,
+        },
+        ".a-submit": {
+            "t-on-click.prevent.stop": this._onSubmitClick,
+            "t-att-disabled": () => !this.selectedSlot,
+        },
+        "li.page-item": {
+            "t-on-click.prevent.stop": this._onChangePageClick,
+        },
+        ".o_wevent_slot_btn": {
+            "t-on-click.prevent.stop": this._onSlotSelected,
+        },
+        ".o_wevent_slot_btn_cancel, .btn-close": {
+            "t-on-click": this._onClose,
+        },
+    };
+
+    get selectedSlot() {
+        return this.form.getAttribute("data-selected-slot-id");
+    }
+
+    setup() {
+        this.currentSlotPage = 0;
+        this.datesLimitPerPage = 6;
+        this.form = this.el.querySelector("#slot_registration_form");
+        this.selectedSlotDatetime = this.el.querySelector(".o_wevent_selected_slot");
+        this.selectedSlotTitle = this.el.querySelector(".o_wevent_selected_slot_title");
+        // Init first slot page
+        if (this.el.querySelector(".pagination")) {
+            this._changePage(0);
+        }
+    }
+
+    /**
+     * Display the specified slot page. Index starting at 0.
+     * @param {Integer} numPage
+     */
+    _changePage(numPage) {
```

Comment:
So from what I understand, you add "d-none" on every single date and then remove the d-none based on the range you want?

That seems a heavily manual and implies a lot of useless nodes in the DOM + a lot of DOM manipulations every time we change dates, especially with a high number of slots.

Wouldn't it be better to assign "data-slots" to a div in the DOM and give it a list of dict {date: [slot1: {start, end}, ...]}.
Then here you'd empty the div when pages are clicked and you'd have a separate JS template that you render with the correct sub-set of dates you want.

Would take some work and maybe not top priority but this is a bit meh.